### PR TITLE
Log error if nginx-proxy is not run

### PR DESCRIPTION
**Pull request by SmotrovaLilit**
I added log error if nginx-proxy is not run,  and i modified function docker_exec in case if docker can not create exec comand.

### CODE REVIEW

File: app/functions.sh [hunk 1]
```diff
@@ -8,6 +8,25 @@
 [[ -z "${END_HEADER:-}" ]] && \
  declare -r END_HEADER='## End of configuration add by letsencrypt container'
 
+function check_nginx_proxy_container_run {
+    local _nginx_proxy_container=$(nginx_proxy_container)
+    if [[ $(docker_api "/containers/${_nginx_proxy_container}/json" | jq -r '.State.Status') = "running" ]];then
+        return 0
+    fi
+
+    echo "$(date), Error: nginx-proxy container ${_nginx_proxy_container}  isn't run" >&2
```
Comment (buchdag) [inline app/functions.sh]: `date` output should be formatted like docker-gen (`date "+%Y/%m/%d %T"`).

Minor english typo too: "nginx-proxy container `id` isn't running."

File: app/functions.sh [hunk 2]
```diff
@@ -59,8 +78,11 @@ function docker_exec {
     local cmd="${2?missing command}"
     local data=$(printf '{ "AttachStdin": false, "AttachStdout": true, "AttachStderr": true, "Tty":false,"Cmd": %s }' "$cmd")
     exec_id=$(docker_api "/containers/$id/exec" "POST" "$data" | jq -r .Id)
-    if [[ -n "$exec_id" ]]; then
+    if [[ -n $exec_id && ($exec_id != "null") ]]; then
         docker_api /exec/$exec_id/start "POST" '{"Detach": false, "Tty":false}'
+    else
+        echo "$(date), Error: can't exec command ${cmd} in container ${id}. Check container is run" >&2
```
Comment (buchdag) [inline app/functions.sh]: `date` output formatting + typo: "Check if the container is running."

File: app/functions.sh [hunk 3]
```diff
@@ -102,6 +124,7 @@ reload_nginx() {
             echo "Reloading nginx proxy (${_nginx_proxy_container})..."
             docker_exec "${_nginx_proxy_container}" \
                         '[ "sh", "-c", "/usr/local/bin/docker-gen /app/nginx.tmpl /etc/nginx/conf.d/default.conf; /usr/sbin/nginx -s reload" ]'
+            [[ $? -eq 1 ]] && echo "$(date), Error: can't reload nginx-proxy. Check container is run" >&2
```
Comment (buchdag) [inline app/functions.sh]: Same as above.

**Comment by buchdag** [comment]
Hi ! Thanks for the PR. Were you able to find exactly why the `check_nginx_proxy_container_run` is failing with the three containers setup ?

File: app/functions.sh (hunk 1 above)
Comment (SmotrovaLilit) [reply app/functions.sh]: Fixed

File: app/functions.sh (hunk 2 above)
Comment (SmotrovaLilit) [reply app/functions.sh]: Fixed

File: app/functions.sh (hunk 3 above)
Comment (SmotrovaLilit) [reply app/functions.sh]: Fixed

**Comment by SmotrovaLilit** [comment]
Hi @buchdag

In case with three containers nginx_proxy_container is empty because function get_nginx_proxy_cid dosn't run: https://github.com/JrCs/docker-letsencrypt-nginx-proxy-companion/blob/master/app/entrypoint.sh#L94

**Comment by buchdag** [comment]
... and the test are currently running with `--volumes-from`, so indeed you don't get an nginx container id on three container setup. Make sense.

I'll cleanup the "get nginx container id" logic later to make this work with the three containers setup too, that was planned anyway.

I've got only two minor request for changes that I missed the first time.

File: app/functions.sh [hunk 4]
```diff
@@ -59,8 +78,11 @@ function docker_exec {
     local cmd="${2?missing command}"
     local data=$(printf '{ "AttachStdin": false, "AttachStdout": true, "AttachStderr": true, "Tty":false,"Cmd": %s }' "$cmd")
     exec_id=$(docker_api "/containers/$id/exec" "POST" "$data" | jq -r .Id)
-    if [[ -n "$exec_id" ]]; then
+    if [[ -n $exec_id && ($exec_id != "null") ]]; then
```
Comment (buchdag) [inline app/functions.sh]: The parenthesis here are not needed.

I'd prefer `[[ -n "$exec_id" && "$exec_id" != "null" ]]` to stay in line with existing code style.

File: app/functions.sh [hunk 5]
```diff
@@ -102,6 +124,7 @@ reload_nginx() {
             echo "Reloading nginx proxy (${_nginx_proxy_container})..."
             docker_exec "${_nginx_proxy_container}" \
                         '[ "sh", "-c", "/usr/local/bin/docker-gen /app/nginx.tmpl /etc/nginx/conf.d/default.conf; /usr/sbin/nginx -s reload" ]'
+            [[ $? -eq 1 ]] && echo "$(date "+%Y/%m/%d %T"), Error: can't reload nginx-proxy. Check if the container is running" >&2
```
Comment (buchdag) [inline app/functions.sh]: I think the "Check if the container is running" on this error message is redundant, as we'll get it from the `docker_exec` error just before that, right ?

File: app/functions.sh [hunk 6]
```diff
@@ -8,6 +8,25 @@
 [[ -z "${END_HEADER:-}" ]] && \
  declare -r END_HEADER='## End of configuration add by letsencrypt container'
 
+function check_nginx_proxy_container_run {
+    local _nginx_proxy_container=$(nginx_proxy_container)
+    if [[ $(docker_api "/containers/${_nginx_proxy_container}/json" | jq -r '.State.Status') = "running" ]];then
+        return 0
+    fi
+
+    echo "$(date "+%Y/%m/%d %T"), Error: nginx-proxy container ${_nginx_proxy_container}  isn't running" >&2
```
Comment (buchdag) [inline app/functions.sh]: Could you end the error messages with a `.` ? 😃

Comment (SmotrovaLilit) [reply app/functions.sh]: Of course)

File: app/functions.sh (hunk 4 above)
Comment (SmotrovaLilit) [reply app/functions.sh]: Fixed.

File: app/functions.sh (hunk 5 above)
Comment (SmotrovaLilit) [reply app/functions.sh]: Yes, you are right.

**Comment by buchdag** [comment]
Thanks for the changes.

Looks good to me, can you squash the three commits into one or do you want me to do it ?

**Comment by SmotrovaLilit** [comment]
Thanks for reviewing. I’ve squashed commits into one

**Pull request merged**